packages/expo-device/android/src/main/java/expo/modules/device/DeviceModule.java

```diff
@@ -0,0 +1,332 @@
+    if (windowManager == null) {
+      return DeviceType.UNKNOWN;
+    }
+
+    // Get display metrics to see if we can differentiate handsets and tablets.
+    // NOTE: for API level 16 the metrics will exclude window decor.
+    DisplayMetrics metrics = new DisplayMetrics();
+    windowManager.getDefaultDisplay().getMetrics(metrics);
+
+    // Calculate physical size.
+    double widthInches = metrics.widthPixels / (double) metrics.xdpi;
+    double heightInches = metrics.heightPixels / (double) metrics.ydpi;
+    double diagonalSizeInches = Math.sqrt(Math.pow(widthInches, 2) + Math.pow(heightInches, 2));
+
+    if (diagonalSizeInches >= 3.0 && diagonalSizeInches <= 6.9) {
+      // Devices in a sane range for phones are considered to be Handsets.
+      return DeviceType.HANDSET;
+    } else if (diagonalSizeInches > 6.9 && diagonalSizeInches <= 18.0) {
+      // Devices larger than handset and in a sane range for tablets are tablets.
+      return DeviceType.TABLET;
+    } else {
+      // Otherwise, we don't know what device type we're on/
+      return DeviceType.UNKNOWN;
+    }
+  }
+
+  @ExpoMethod
+  public void getSerialNumberAsync(final Promise promise) {
+
+    int permissionCheck = ContextCompat.checkSelfPermission(mContext, Manifest.permission.READ_PHONE_STATE);
+    if (permissionCheck == PackageManager.PERMISSION_GRANTED) {
+      try {
+        if (android.os.Build.VERSION.SDK_INT < 26) {
+          promise.resolve(Build.SERIAL);
+        } else {
+          promise.resolve(Build.getSerial());
+        }
+      } catch (SecurityException e) {
+        Log.e(TAG, e.getMessage());
+        promise.reject(e);
+      }
+    } else {
+      Permissions mPermissions = mModuleRegistry.getModule(Permissions.class);
+      String permissionsTable[] = {Manifest.permission.READ_PHONE_STATE};
+      mPermissions.askForPermissions(permissionsTable, new Permissions.PermissionsRequestListener() {
+        @Override
+        public void onPermissionsResult(int[] results) {
+          if (results[0] == PackageManager.PERMISSION_GRANTED) {
+            try {
+              if (android.os.Build.VERSION.SDK_INT < 26) {
+                promise.resolve(Build.SERIAL);
+              } else {
+                promise.resolve(Build.getSerial());
+              }
+            } catch (SecurityException e) {
+              Log.e(TAG, e.getMessage());
+              promise.reject(e);
+            }
+          } else {
+            promise.reject(new SecurityException("User rejected permissions"));
+          }
+        }
+      });
+    }
+  }
+
+  @ExpoMethod
+  public void getFreeDiskStorageAsync(Promise promise) {
+    try {
+      StatFs external = new StatFs(Environment.getExternalStorageDirectory().getAbsolutePath());
+      long availableBlocks;
+      long blockSize;
+
+      availableBlocks = external.getAvailableBlocksLong();
+      blockSize = external.getBlockSizeLong();
```

Comment:
you can define these variables at the same time you declare them, on line 215-216